view.php

```diff
@@ -47,8 +47,31 @@
 
     echo $OUTPUT->header();
 
+    $entries = $DB->get_records('cpdlogbook_entries', ['cpdlogbookid' => $cm->instance]);
+    $sum = 0;
+    foreach ($entries as $e) {
+        $sum += $e->points;
+    }
+
+    // Output the points as a ratio, ie. 5 / 20.
+    $a = new stdClass();
+    $a->sum = $sum;
+    $a->required = $record->totalpoints;
+    echo html_writer::tag('p', get_string('pointratio', 'mod_cpdlogbook', $a), ['class' => 'h2']);
+
+    // Output the points as a progress bar towards completion.
+    echo html_writer::div(
+        html_writer::div(
+            '',
+            'progress-bar',
+            ['style' => 'width: '.(100 * $sum / $record->totalpoints).'%']
```

Comment:
This is great. Some very minor feedback, as a general try to split your logic from the display code. So things like the 100 * sum should be calculated earlier. It should also have some max 100% logic in there.

In terms of the UI, the only small thing is you could add the percentage as a number inside the markup, and the other thing that could be good is to add the class to make is a green 'success' bar chart.

There is a lot of the primary blue color.

![image](https://user-images.githubusercontent.com/187449/130737714-08c8b93b-4bda-46c5-9277-cda24ed3489d.png)